typescript/src/hooks/directionalFeeHook.ts

```diff
@@ -0,0 +1,94 @@
+import { HookBase } from './types';
+import { MathSol } from '../utils/math';
+import { SwapInput } from '@/vault/types';
+
+
+export type HookStateDirectionalFee = {
+    tokens: string[];
+    balancesLiveScaled18: bigint[];
+}
+
+export class DirectionalFeeHook implements HookBase {
+    public shouldCallComputeDynamicSwapFee = true;
+    public shouldCallBeforeSwap = false;
+    public shouldCallAfterSwap = false;
+    public shouldCallBeforeAddLiquidity = false;
+    public shouldCallAfterAddLiquidity = false;
+    public shouldCallBeforeRemoveLiquidity = false;
+    public shouldCallAfterRemoveLiquidity = false;
+    public enableHookAdjustedAmounts = false;
+
+
+    onComputeDynamicSwapFee(
+        params: SwapInput,
+        staticSwapFeePercentage: bigint,
+        hookState: HookStateDirectionalFee
+    ): { success: boolean; dynamicSwapFee: bigint } {
+
+        const lastBalancesLiveScaled18 = hookState.balancesLiveScaled18;
+
+        const tokenInIndex = hookState.tokens.indexOf(params.tokenIn);
+        const tokenOutIndex = hookState.tokens.indexOf(params.tokenOut);
+
+        const calculatedSwapFeePercentage = this.calculateExpectedSwapFeePercentage(
+            lastBalancesLiveScaled18,
+            params.amountRaw, // does this need to be scaled? 
```

Comment:
Should be able to just replicate the SC approach. So if its scaled in SC it needs scaled here. Let me know if this doesn't work.

Reply:
Scaling has been added.